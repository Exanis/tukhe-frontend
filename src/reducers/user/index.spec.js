import axios from 'axios';
import slice, { selectUser } from './index';

it('Should have correct reducers', () => {
    const state = {
        token: false,
        dashboards: [],
        currentDashboard: null,
        widgets: []
    };
    const action = {
        type: slice.actions.setToken,
        payload: 'test'
    };

    jest.spyOn(Storage.prototype, 'setItem');
    expect(slice.reducer(state, action)).toMatchSnapshot();
    expect(Storage.prototype.setItem).toHaveBeenCalledWith('tukhe-token', 'test');
    expect(axios.defaults.headers.common).toMatchSnapshot();

    action.payload = null;
    expect(slice.reducer(state, action)).toMatchSnapshot();
    expect(axios.defaults.headers.common).toEqual({});

    action.type = slice.actions.setDashboards;
    action.payload = 'test';
    expect(slice.reducer(state, action)).toMatchSnapshot();

    action.type = slice.actions.selectDashboard;
    expect(slice.reducer(state, action)).toMatchSnapshot();

    action.type = slice.actions.setWidgets;
    expect(slice.reducer(state, action)).toMatchSnapshot();
});

it('Should correctly select user', () => {
    const state = {
        user: 'test',
        error: 'bad'
    };

    expect(selectUser(state)).toEqual('test');
});
