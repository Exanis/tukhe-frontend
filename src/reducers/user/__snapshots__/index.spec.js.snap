// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have correct reducers 1`] = `
Object {
  "currentDashboard": null,
  "dashboards": Array [],
  "token": "test",
  "widgets": Array [],
}
`;

exports[`Should have correct reducers 2`] = `
Object {
  "Authorization": "Token test",
}
`;

exports[`Should have correct reducers 3`] = `
Object {
  "currentDashboard": null,
  "dashboards": Array [],
  "token": null,
  "widgets": Array [],
}
`;

exports[`Should have correct reducers 4`] = `
Object {
  "currentDashboard": null,
  "dashboards": "test",
  "token": false,
  "widgets": Array [],
}
`;

exports[`Should have correct reducers 5`] = `
Object {
  "currentDashboard": "test",
  "dashboards": Array [],
  "token": false,
  "widgets": Array [],
}
`;

exports[`Should have correct reducers 6`] = `
Object {
  "currentDashboard": null,
  "dashboards": Array [],
  "token": false,
  "widgets": "test",
}
`;
